### PR TITLE
DVE-2017-0009: axc.nl bogus DNSSEC denial of existence

### PR DESCRIPTION
**Pull request by vdukhovni**

### CODE REVIEW

**Comment by jelu** [comment]
Are they using the same software so this could be directed against the software instead of making one for each domain?

File: 2017/DVE-2017-0009.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+This breaks email delivery from DANE-enabled sending MTAs, since
+bogus TLSA denial of existence is indistinguishable from an active
+downgrade attack.
+
+## Evidence
+
+(RRSIG records and other extraneous dig output elided)
+
+#### NODATA should be NXDOMAIN:
```
Comment (jelu) [inline 2017/DVE-2017-0009.md]: Header level is wrong, should be 3 not 4.

File: 2017/DVE-2017-0009.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+    @nszero2.axc.nl.[159.253.2.101]
+    ; <<>> DiG 9.10.4-P2 <<>> +dnssec +noall +cmd +comment +qu +ans +auth +nocl +nottl +nosplit +norecur -t tlsa _25._tcp.cameras-kopen.nl @159.253.2.101
+    ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 30262
+    ;; flags: qr aa; QUERY: 1, ANSWER: 0, AUTHORITY: 4, ADDITIONAL: 1
+    ;_25._tcp.cameras-kopen.nl. IN TLSA
+    cameras-kopen.nl.       SOA     nszero1.axc.nl. hostmaster.cameras-kopen.nl.  2010041707 14400 3600 1209600 86400
+    cameras-kopen.nl.       NSEC    cameras-kopen.nl. A NS SOA RRSIG NSEC DNSKEY
+
+#### NSEC chain should include wildcard:
```
Comment (jelu) [inline 2017/DVE-2017-0009.md]: Same here.

File: 2017/DVE-2017-0009.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+    @nszero2.axc.nl.[159.253.2.101]
+    ; <<>> DiG 9.10.4-P2 <<>> +dnssec +noall +cmd +comment +qu +ans +auth +nocl +nottl +nosplit +norecur -t tlsa _25._tcp.smtp.mwalet.nl @159.253.2.101
+    ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 30236
+    ;; flags: qr aa; QUERY: 1, ANSWER: 0, AUTHORITY: 4, ADDITIONAL: 1
+    ;_25._tcp.smtp.mwalet.nl. IN TLSA
+    mwalet.nl.              SOA     nszero1.axc.nl. hostmaster.mwalet.nl.  2016120104 14400 3600 1209600 86400
+    mwalet.nl.              NSEC    mwalet.nl. A NS SOA MX TXT RRSIG NSEC DNSKEY
+
+#### List of known affected TLSA qnames
```
Comment (jelu) [inline 2017/DVE-2017-0009.md]: And here.

File: 2017/DVE-2017-0010.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
+# DVE-2017-0009: infracom.nl bogus DNSSEC denial of existence
```
Comment (jelu) [inline 2017/DVE-2017-0010.md]: Wrong number in the title.

File: 2017/DVE-2017-0010.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+This breaks email delivery from DANE-enabled sending MTAs, since
+bogus TLSA denial of existence is indistinguishable from an active
+downgrade attack.
+
+## Evidence
+
+(RRSIG records and other extraneous dig output elided)
+
+#### Incorrect wildcard NSEC next:
```
Comment (jelu) [inline 2017/DVE-2017-0010.md]: Header level.

File: 2017/DVE-2017-0010.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    ; <<>> DiG 9.10.4-P2 <<>> +dnssec +noall +cmd +comment +qu +ans +auth +nocl +nottl +nosplit +norecur -t tlsa _25._tcp.fivelpoort.nu @94.247.75.5
+    ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 38825
+    ;; flags: qr aa; QUERY: 1, ANSWER: 2, AUTHORITY: 6, ADDITIONAL: 1
+    ;_25._tcp.fivelpoort.nu.        IN TLSA
+    _25._tcp.fivelpoort.nu. CNAME   fivelpoort.nu.
+    *.fivelpoort.nu.        NSEC    *.fivelpoort.nu. CNAME RRSIG NSEC
+    fivelpoort.nu.          SOA     ns3.infracom.nl. dom-reg.infracom.nl. 2016071802 14400 7200 86400 3600
+    fivelpoort.nu.          NSEC    *.fivelpoort.nu. A NS SOA AAAA RRSIG NSEC DNSKEY
+
+#### List of known affected TLSA qnames
```
Comment (jelu) [inline 2017/DVE-2017-0010.md]: And here.

**Review by jelu — DISMISSED** [review]

**Comment by jelu** [comment]
To make it easier to manage I think you should move DVE-2017-0010 to another pull request, unless we merge them into a singular issue on the software.

File: 2017/DVE-2017-0009.md (hunk 1 above)
Comment (vdukhovni) [reply 2017/DVE-2017-0009.md]: Level 3 looks much too large IMHO, how important is it to have strictly sequential levels?

File: 2017/DVE-2017-0010.md (hunk 4 above)
Comment (vdukhovni) [reply 2017/DVE-2017-0010.md]: Thanks, will push a fix for the copy/paste error.

**Comment by vdukhovni** [comment]
> @jelu Are they using the same software so this could be directed against the software instead of making one for each domain?

I don't know what software they are using. Based on the somewhat different symptoms, I'd say not.
I did initially create a PR with just one DVE, but after a while I wanted to add another and did not want to spin up another branch and possibly out of order assignments. Perhaps for now these could go through together.  I'll keep these separate going forward.

**Comment by vdukhovni** [comment]
Ping... What now?

File: 2017/DVE-2017-0009.md (hunk 1 above)
Comment (oerdnj) [reply 2017/DVE-2017-0009.md]: It's important if we are ever to create any automated web page output from the submitted issues.

**Comment by oerdnj** [comment]
@vdukhovni Please fix the header levels  (for future automatic parsing) and I think I  can accept it as is. I am fine with two DVEs in one PR.

**Comment by vdukhovni** [comment]
Header levels "fixed". This does not improve visual appearance, but so be it...

**Comment by jelu** [comment]
@vdukhovni The out of order assignment may exactly happen when you do it like this since we can use the PR title to see where we are in the numbering. And you should start using branch based PR's and not work from the master branch, not using upstream branches in your fork makes life much easier being in sync with the upstream.

**Comment by vdukhovni** [comment]
This should be ready to merge.

**Comment by vdukhovni** [comment]
What's the next step?

**Comment by jelu** [comment]
@vdukhovni To have patience and wait for @oerdnj to approve or anyone else of the @DNS-OARC/dve-maintainers (*poke*)

**Comment by vdukhovni** [comment]
IMHO the process for getting changes merged ought to be substantially more lightweight.  One should not have to work this hard to report an issue.  I'm volunteering my time to report a problem, not proposing changes to mission-critical code... The perfect is the enemy of the good.

**Comment by jelu** [comment]
@oerdnj ping?

**Review by oerdnj — APPROVED** [review]
I would rather split this into two next time, but let's go ahead now.

**Pull request merged**